### PR TITLE
Independent sample analysis (2 of 2)

### PR DESCRIPTION
**Pull request by jashapiro**
#### Purpose/implementation

Generate files of independent samples for downstream analysis where more than one samples from the same indidvidual would bias analysis or is otherwise not desirable

#### Issue

#155 

#### Directions for reviewers

Do the sample lists look reasonable (truly independent?)
Should this PR include adding  canonical versions of these files elsewhere in the project? If so, where? Should they be part of the data download?

#### Results

The script generates 4 files of independent samples: one with all primary tumors and WGS sequences, as well as files that include WXS samples and/or non-primary tumors.

In summary, the independent spain lists contain: 
641 WGS primary specimens
788 WGS specimens (including non-primary)
657 WGS+WXS primary specimens
804 WGS+WXS specimens (including non-primary)

#### Docker and continuous integration

_Check all those that apply or remove this section if it is not applicable._

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

### CODE REVIEW

**Comment by jharenza** [comment]
> I had one question to check my understanding. I did pull this locally and kick the tires a bit. Everything was as expected/described as far as the sample sets go.
> 
> > Should this PR include adding canonical versions of these files elsewhere in the project? If so, where? Should they be part of the data download?
> 
> I am a big fan of including this as part of the download. Is this something we could potentially do as part of #146 @jharenza?

How do you envision this in the download? A separate clinical file or as another column in the clinical file for which samples to use? I am working on adding a lot more info to the clinical file, and we also have two samples which need to be pulled due to non-consent, so I would suggest not as another file, but we could add a column if you give me the final list?

File: analyses/independent-samples/01-generate-independent-specimens.R [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+# Generate WGS independent samples
+wgs_samples <- tumor_samples %>%
+  dplyr::filter(experimental_strategy == "WGS")
+
+wgs_primary <- independent_samples(wgs_samples, tumor_types = "primary")
+wgs_primary_plus <- independent_samples(wgs_samples, tumor_types = "prefer_primary")
+
+# Generate lists for WXS only samples
+wxs_only_samples <-  tumor_samples %>% 
+  dplyr::filter(!(Kids_First_Participant_ID %in%
```
[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/independent-samples/01-generate-independent-specimens.R]: I would prefer if we could retain the completed set of WXS only samples rather than the limited WGS for these patients, since we are looking at coding regions for the majority of SNV analyses and so that we do not lose a large chunk of DIPG samples - one of the currently incurable tumors. For CNV/SV analyses, we have only called those on WGS, so it makes sense to prefer WGS there.

[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/independent-samples/01-generate-independent-specimens.R]: Or, do you just want them as file lists of PT and BS ids, which would be pretty easy to add?

[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/independent-samples/01-generate-independent-specimens.R]: There are 4 files, two of which have WGS + WXS for the samples where WGS is not available, so that use case should be covered. Do you think there is a reason to prefer WXS? Do we want a WXS-preferred list too?

**Comment by jashapiro** [comment]
> How do you envision this in the download? A separate clinical file or as another column in the clinical file for which samples to use? 

I would think having a separate file(s) is probably the easiest. You can run the `01-generate-independent-specimens.R` script and include the output.

File: analyses/independent-samples/01-generate-independent-specimens.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/independent-samples/01-generate-independent-specimens.R]: OK, that works! I think typically, I would say we could use one or the other as a validation cohort, but that is probably beyond the scope currently. Was thinking WXS preferred due to higher coverage and more TP calls, however, we do at least know the drivers are consistently called in WXS/WGS for that cohort. We can probably leave as is for now?

**Comment by jharenza** [comment]
> > How do you envision this in the download? A separate clinical file or as another column in the clinical file for which samples to use?
> 
> I would think having a separate file(s) is probably the easiest. You can run the `01-generate-independent-specimens.R` script and include the output.

OK - I can provide these as lists in the release so people can extract those samples from the clinical file (since so many clinical file edits are upcoming). Once the PR is merged, can you point me to the output files? I will also have to remove two newly-annotated as non-consent samples. I haven't yet set up docker for this and to do so will require some tutorial-ing on my end and have a pressing deadline nov 15, so won't be able to do this until after then.
